### PR TITLE
Lmp.cfg common (lmp/lmp-mfgtool)

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

**Review by MrCry0 — APPROVED** [review]
LGTM, not tested.

**Review by quaresmajose — APPROVED** [review]
LGTM

**Comment by ldts** [comment]
Nice!

**Review by angolini — APPROVED** [review]

**Review by MrCry0 — APPROVED** [review]
LGTM

**Comment by igoropaniuk** [comment]
@ricardosalveti rebased in the latest `main`

**Comment by mike-scott** [comment]
@igoropaniuk Thank you for pushing this.  I'll drop my very old PR:
https://github.com/foundriesio/meta-lmp/pull/276 where I tried the same thing and never finished.

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-fio/lmp-common.cfg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CONFIG_FIT=y
+CONFIG_USE_BOOTCOMMAND=y
+CONFIG_BOOTDELAY=-2
+CONFIG_FIT_VERBOSE=y
+# CONFIG_DISTRO_DEFAULTS is not set
+CONFIG_FIT_SIGNATURE=y
+CONFIG_FIT_SIGNATURE_STRICT=y
+CONFIG_RSA=y
```
[the remark this answers is not in the document]
Comment (igoropaniuk) [reply meta-lmp-base/recipes-bsp/u-boot/u-boot-fio/lmp-common.cfg:8]: I think we can add those in a separate PR.
The idea of this PR is to move existing common config flags to lmp-common with minimal side effects

**Comment by ricardosalveti** [comment]
Comparing configs, beagle:

```
rsalveti@evatuf:~/build/lmp/build-lmp/tmp-lmp/work/beaglebone_yocto-lmp-linux-gnueabi/u-boot-fio/2022.04-r0/build$ diff -Naur /tmp/u-boot/u-boot-fio-config-beaglebone-yocto .config
--- /tmp/u-boot/u-boot-fio-config-beaglebone-yocto      2023-04-03 14:07:49.000000000 -0300
+++ .config     2023-04-03 21:49:40.179232422 -0300
@@ -275,7 +275,7 @@
 # CONFIG_LTO is not set
 CONFIG_CC_HAS_ASM_INLINE=y
 # CONFIG_XEN is not set
-CONFIG_DISTRO_DEFAULTS=y
+# CONFIG_DISTRO_DEFAULTS is not set
 CONFIG_ENV_VARS_UBOOT_CONFIG=y
 # CONFIG_SYS_BOOT_GET_CMDLINE is not set
 # CONFIG_SYS_BOOT_GET_KBD is not set
@@ -316,7 +316,7 @@
 CONFIG_FIT_SIGNATURE=y
 CONFIG_FIT_SIGNATURE_MAX_SIZE=0x10000000
 # CONFIG_FIT_RSASSA_PSS is not set
-# CONFIG_FIT_SIGNATURE_STRICT is not set
+CONFIG_FIT_SIGNATURE_STRICT=y
 # CONFIG_FIT_CIPHER is not set
 CONFIG_FIT_VERBOSE=y
 # CONFIG_FIT_BEST_MATCH is not set
@@ -369,7 +369,7 @@
 # Autoboot options
 #
 CONFIG_AUTOBOOT=y
-CONFIG_BOOTDELAY=2
+CONFIG_BOOTDELAY=-2
 # CONFIG_AUTOBOOT_KEYED is not set
 # CONFIG_AUTOBOOT_USE_MENUKEY is not set
 # CONFIG_USE_BOOTARGS is not set
```

I don't think we set UBOOT_SIGN_ENABLE for beagle (STRICT), and bootdelay is set to 2 because we don't yet have a fully supported boot command (like we do for imx), so we prefer to give the option to the user to use the prompt.

Distro defaults can probably be disabled, but might impact as well.

**Review by angolini — APPROVED** [review]
LGTM

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/sun8i/lmp.cfg [hunk 2]
```diff
@@ -1,8 +1,7 @@
...
-CONFIG_FIT_VERBOSE=y
 CONFIG_BOOTCOUNT_LIMIT=y
 CONFIG_BOOTCOUNT_ENV=y
 CONFIG_BOOTCOUNT_BOOTLIMIT=3
-CONFIG_BOOTDELAY=-2
-CONFIG_USE_BOOTCOMMAND=y
 CONFIG_BOOTCOMMAND="fatload mmc 0:1 ${scriptaddr} /boot.itb; source ${scriptaddr}; env default -a; saveenv; reset"
+# CONFIG_FIT_SIGNATURE_STRICT is not set
+CONFIG_BOOTDELAY=2
+CONFIG_DISTRO_DEFAULTS=y
```
Comment (ricardosalveti) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/sun8i/lmp.cfg:7]: We can potentially have these 3 options in another fragment (lmp-common-no-security or similar), and have it included via a conditional (UBOOT_SIGN_ENABLE basically).

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-fio-mfgtool/lmp-common.cfg [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+# CONFIG_CMD_GETTIME is not set
+# CONFIG_CMD_POWEROFF is not set
+# CONFIG_CMD_RTC is not set
+# CONFIG_CMD_SNTP is not set
+# CONFIG_CMD_TIME is not set
+# CONFIG_CMD_TIMER is not set
+# CONFIG_DFU is not set
+# CONFIG_ENV_IS_IN_SPI_FLASH is not set
+CONFIG_SPL_FIT_SIGNATURE_STRICT=y
+# CONFIG_SPL_MMC is not set
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-bsp/u-boot/u-boot-fio-mfgtool/lmp-common.cfg:91]: Nit, can we sort the options to gather the SPL related ones together? Easier to read and potentially debug later (which are usually the options we want to carefully review due size restrictions).

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/sun8i/lmp.cfg (hunk 2 above)
Comment (igoropaniuk) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/sun8i/lmp.cfg]: done, thanks!

**Comment by igoropaniuk** [comment]
@ricardosalveti all comments addressed

**Comment by ricardosalveti** [comment]
@igoropaniuk mind doing a final rebase, including the changes for imx8ulp and imx93? We should be good to merge after this, now that v90 was released.

Sorry for the constant rebase requests.

**Comment by igoropaniuk** [comment]
@ricardosalveti done

**Comment by igoropaniuk** [comment]
As we discussed offline earlier, I'll also adjust lmp.cry for imx8ulp and imx93 in this PR.

**Review by MrCry0 — APPROVED** [review]

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx93-11x11-lpddr4x-evk/.lmp.cfg.swp
Comment (ricardosalveti) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx93-11x11-lpddr4x-evk/.lmp.cfg.swp:1]: Please remove this one, pushed by mistake.

**Comment by ricardosalveti** [comment]
imx8mn-ddr4-evk / imx8mn-ddr4-evk-sec (might have size issues):
```
--- /tmp/u-boot/old/u-boot-fio-mfgtool-config-imx8mn-ddr4-evk   2023-06-12 21:00:08.000000000 -0300
+++ imx8mn-ddr4-evk/u-boot-fio-mfgtool-config-imx8mn-ddr4-evk   2023-06-14 16:29:01.792400569 -0300
@@ -377,9 +377,11 @@
 CONFIG_FIT_PRINT=y
 CONFIG_SPL_FIT=y
 # CONFIG_SPL_FIT_PRINT is not set
-# CONFIG_SPL_FIT_FULL_CHECK is not set
-# CONFIG_SPL_FIT_SIGNATURE is not set
-# CONFIG_SPL_FIT_SIGNATURE_STRICT is not set
+CONFIG_SPL_FIT_FULL_CHECK=y
+CONFIG_SPL_FIT_SIGNATURE=y
+CONFIG_SPL_FIT_SIGNATURE_MAX_SIZE=0x10000000
+# CONFIG_SPL_FIT_RSASSA_PSS is not set
+CONFIG_SPL_FIT_SIGNATURE_STRICT=y
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_SPL_LOAD_FIT_ADDRESS=0x0
 # CONFIG_SPL_LOAD_FIT_APPLY_OVERLAY is not set
@@ -571,12 +573,11 @@
 # CONFIG_SPL_USB_GADGET is not set
 CONFIG_SPL_WATCHDOG=y
 # CONFIG_SPL_YMODEM_SUPPORT is not set
-CONFIG_SPL_ATF=y
-# CONFIG_SPL_ATF_LOAD_IMAGE_V2 is not set
-# CONFIG_SPL_ATF_NO_PLATFORM_PARAM is not set
+# CONFIG_SPL_ATF is not set
 # CONFIG_SPL_AM33XX_ENABLE_RTC32K_OSC is not set
 # CONFIG_SPL_OPTEE_IMAGE is not set
 CONFIG_IMAGE_SIGN_INFO=y
+CONFIG_SPL_IMAGE_SIGN_INFO=y
 # CONFIG_FDT_SIMPLEFB is not set

 #
@@ -1781,6 +1782,7 @@
 # CONFIG_ECDSA is not set
 CONFIG_RSA=y
 CONFIG_SPL_RSA=y
+CONFIG_SPL_RSA_VERIFY=y
 CONFIG_RSA_VERIFY=y
 # CONFIG_RSA_VERIFY_WITH_PKEY is not set
 CONFIG_RSA_SOFTWARE_EXP=y
```

u-boot-fio-mfgtool fails to build for imx8mn-lpddr4-evk/imx8mn-lpddr4-evk-sec.

**Comment by igoropaniuk** [comment]
Completely missed testing imx8mn-lpddr4-evk/imx8mn-lpddr4-evk-sec

I've fixed all the issues @ricardosalveti

**Review by ricardosalveti — APPROVED** [review]
LGTM, now for more testing :-)

**Pull request merged**